Framework/CurveFitting/test/Algorithms/FitTest.h

```diff
@@ -572,6 +575,220 @@ class FitTest : public CxxTest::TestSuite {
+         4.242424, 4.343434, 4.444444, 4.545455, 4.646465, 4.747475, 4.848485,
+         4.949495, 5.050505, 5.151515, 5.252525, 5.353535, 5.454545, 5.555556,
+         5.656566, 5.757576, 5.858586, 5.959596, 6.060606, 6.161616, 6.262626,
+         6.363636, 6.464646, 6.565657, 6.666667, 6.767677, 6.868687, 6.969697,
+         7.070707, 7.171717, 7.272727, 7.373737, 7.474747, 7.575758, 7.676768,
+         7.777778, 7.878788, 7.979798, 8.080808, 8.181818, 8.282828, 8.383838,
+         8.484848, 8.585859, 8.686869, 8.787879, 8.888889, 8.989899, 9.090909,
+         9.191919, 9.292929, 9.393939, 9.494949, 9.595960, 9.696970, 9.797980,
+         9.898990};
+    Y = {0.000000, 0.000000, 0.000000, 0.000000, 0.000001, 0.000001, 0.000002,
+         0.000004, 0.000006, 0.000012, 0.000021, 0.000036, 0.000063, 0.000108,
+         0.000183, 0.000305, 0.000503, 0.000818, 0.001314, 0.002084, 0.003262,
+         0.005041, 0.007692, 0.011586, 0.017229, 0.025295, 0.036664, 0.052465,
+         0.074121, 0.103380, 0.142353, 0.193520, 0.259728, 0.344147, 0.450195,
+         0.581418, 0.741323, 0.933166, 1.159690, 1.422842, 1.723466, 2.061013,
+         2.433271, 2.836167, 3.263660, 3.707743, 4.158590, 4.604836, 5.034009,
+         5.433072, 5.789067, 6.089806, 6.324555, 6.484675, 6.564144, 6.559937,
+         6.472215, 6.304315, 6.062539, 5.755762, 5.394893, 4.992230, 4.560768,
+         4.113514, 3.662855, 3.220017, 2.794656, 2.394584, 2.025646, 1.691721,
+         1.394844, 1.135414, 0.912461, 0.723946, 0.567061, 0.438516, 0.334790,
+         0.252343, 0.187776, 0.137950, 0.100055, 0.071644, 0.050648, 0.035348,
+         0.024356, 0.016568, 0.011127, 0.007378, 0.004829, 0.003121, 0.001991,
+         0.001254, 0.000780, 0.000479, 0.000290, 0.000174, 0.000103, 0.000060,
+         0.000034};
+    E.assign(nY, 1.0);
+
+    X.back() = X[98] + dx;
+    AnalysisDataService::Instance().add("ResolutionTest_WS", ws);
+
+    Algorithms::Fit fit;
+    fit.initialize();
+
+    fit.setPropertyValue(
+        "Function", "composite=Convolution,"
+                    "FixResolution=true,NumDeriv=true;name=Resolution,FileName="
+                    "\"ResolutionTestResolution.res\","
+                    "WorkspaceIndex=0;name=ResolutionTest_Gauss,c=5,h=2,s=1");
+    fit.setPropertyValue("InputWorkspace", "ResolutionTest_WS");
+    fit.setPropertyValue("WorkspaceIndex", "0");
+    fit.execute();
+  }
+
+  void getStretchExpMockData(Mantid::MantidVec &y, Mantid::MantidVec &e) {
+    // values extracted from y(x)=2*exp(-(x/4)^0.5)
+    y = {2,          1.2130613,  0.98613738, 0.84124005, 0.73575888,
+         0.65384379, 0.58766531, 0.53273643, 0.48623347, 0.44626032,
+         0.41148132, 0.38092026, 0.35384241, 0.32968143, 0.30799199,
+         0.28841799, 0.27067057, 0.25451242, 0.2397465,  0.22620756};
+
+    std::transform(y.begin(), y.end(), e.begin(),
+                   std::bind2nd(std::multiplies<double>(), 0.1));
```

Comment:
`std::bind2nd` has been deprecated in favor of `std::bind` and lambda expressions.